jobsprofiler: store DistSQL diagram of jobs in job info

adityamaru opened the pull request:
This change teaches import, cdc, backup and restore
to store their DistSQL plans in the job_info table
under a timestamped info key. The generation and writing
of the plan diagram is done asynchronously so as to not
slow down the execution of the job. A new plan will be
stored everytime the job sets up its DistSQL flow.

Release note: None
Epic: [CRDB-8964](https://cockroachlabs.atlassian.net/browse/CRDB-8964)
Informs: #99729

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/99522)
<!-- Reviewable:end -->

adityamaru committed:
jobsprofiler: store DistSQL diagram of jobs in job info

This change teaches import, cdc, backup and restore
to store their DistSQL plans in the job_info table
under a timestamped info key. The generation and writing
of the plan diagram is done asynchronously so as to not
slow down the execution of the job. A new plan will be
stored everytime the job sets up its DistSQL flow.

Release note: None
Epic: CRDB-8964

dt commented on pkg/jobs/jobsprofiler/profiler.go:
```diff
@@ -0,0 +1,54 @@
+				[]byte(diagURL.String()))
+		})
+		if err != nil {
+			log.Infof(ctx, "failed to generate and write DistSQL diagram for job %d: %v", jobID, err.Error())
```
I'd upgrade this to warning.

dt commented on pkg/jobs/jobsprofiler/profiler.go:
```diff
@@ -0,0 +1,54 @@
+// are done asynchronously and this method does not block on their completion.
+func StorePlanDiagram(
+	ctx context.Context, stopper *stop.Stopper, p *sql.PhysicalPlan, db isql.DB, jobID jobspb.JobID,
+) error {
```
nit: consider just logging and swallowing the error (which never happens) from stopper instead of making this return an error on paper that every caller needs then handle (but never will) since they're just going to log it anyway.

adityamaru commented:
TFTR!

bors r=dt
